website/website/add_youtuber.py

```diff
@@ -0,0 +1,133 @@
+import os
+from urllib.parse import unquote, urlparse
+
+from googleapiclient.discovery import build
+
+
+class YoutubeApi:
+    """A class to interact with the YouTube API."""
+
+    #: (str): The API key used to authenticate requests to the YouTube API.
```

Comment:
AFAICS, all the class attributes has str() type.

I'd consider setting the default value as '' (empty string) instead of None.

If som you can be sure that these attribute values are strings, and work with them as with strings.
No need to check for "is None" or "is not None" in order to process working with string...